backend/danswer/connectors/sharepoint/connector.py

```diff
@@ -75,44 +78,25 @@ def __init__(
     ) -> None:
         self.batch_size = batch_size
         self.graph_client: GraphClient | None = None
-        self.site_data = self.extract_site_and_folder(sites)
+        self.site_data = self._extract_site_and_folder(sites)
 
     @staticmethod
-    def extract_site_and_folder(site_urls: list[str]) -> list[SiteData]:
+    def _extract_site_and_folder(site_urls: list[str]) -> list[SiteData]:
         site_data_list = []
         for url in site_urls:
             parts = url.strip().split("/")
             if "sites" in parts:
                 sites_index = parts.index("sites")
-                site_url = "/".join(parts[:sites_index + 2])
-                folder = parts[sites_index + 2] if len(parts) > sites_index + 2 else None
-                site_data_list.append(SiteData(url=site_url, folder=folder, siteobjects=[], driveitems=[]))
+                site_url = "/".join(parts[: sites_index + 2])
+                folder = (
+                    parts[sites_index + 2] if len(parts) > sites_index + 2 else None
+                )
+                site_data_list.append(
+                    SiteData(url=site_url, folder=folder, siteobjects=[], driveitems=[])
+                )
         return site_data_list
 
-    def load_credentials(self, credentials: dict[str, Any]) -> dict[str, Any] | None:
-        aad_client_id = credentials["aad_client_id"]
-        aad_client_secret = credentials["aad_client_secret"]
-        aad_directory_id = credentials["aad_directory_id"]
-
-        def _acquire_token_func() -> dict[str, Any]:
-            """
-            Acquire token via MSAL
-            """
-            authority_url = f"https://login.microsoftonline.com/{aad_directory_id}"
-            app = msal.ConfidentialClientApplication(
-                authority=authority_url,
-                client_id=aad_client_id,
-                client_credential=aad_client_secret,
-            )
-            token = app.acquire_token_for_client(
-                scopes=["https://graph.microsoft.com/.default"]
-            )
-            return token
-
-        self.graph_client = GraphClient(_acquire_token_func)
-        return None
-
-    def get_all_driveitem_objects(
+    def _get_all_driveitem_objects(
```

Comment:
Follow the underscore convention since this should not be used outside of this module/class. Also functions should not have unanticipated side effects. ie _get_all_driveitem_objects should return driveitem_objects with no side effects if possible. Ask some LLM about "functional programing" and "pure functions". So _get_all_driveitem_objects should not directly modify self unless there is some strong reason for it.